sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java

```diff
@@ -465,7 +466,7 @@ private static DefaultVolumeSpec convertVolume(
                 rawVolume.getSize(),
                 volumeTypeEnum,
                 rawVolume.getPath(),
-                role,
+                ResourceUtils.getRole(preReservedRole, role),
```

Comment:
To avoid mixing up one `String role` with another `String role`, there could be a `Role` class/object that has to be passed around, which would contain the combination of the original `String role` + `String preReservedRole`. This would turn it into a compile-time error if something is expecting a preprocessed role (as manifested by a `Role` object).